specification/hdinsight/resource-manager/Microsoft.HDInsight/HDInsightOnAks/preview/2023-11-01-preview/hdinsight.json

```diff
@@ -0,0 +1,4234 @@
+        },
+        "numReplicas": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The number of task managers."
+        },
+        "jobManager": {
+          "description": "Job Manager container/ process CPU and memory requirements",
+          "$ref": "#/definitions/ComputeResourceDefinition"
+        },
+        "historyServer": {
+          "description": "History Server container/ process CPU and memory requirements",
+          "$ref": "#/definitions/ComputeResourceDefinition"
+        },
+        "taskManager": {
+          "description": "Task Manager container/ process CPU and memory requirements",
+          "$ref": "#/definitions/ComputeResourceDefinition"
+        },
+        "catalogOptions": {
+          "$ref": "#/definitions/FlinkCatalogOptions"
+        },
+        "deploymentMode": {
+          "description": "A string property that indicates the deployment mode of Flink cluster. It can have one of the following enum values => Application, Session. Default value is Session",
+          "type": "string",
+          "enum": [
+            "Application",
+            "Session"
+          ],
+          "x-ms-enum": {
+            "name": "deploymentMode",
+            "modelAsString": true
+          }
+        },
+        "jobSpec": {
+          "$ref": "#/definitions/FlinkJobProfile"
+        }
+      }
+    },
+    "FlinkJobProfile": {
+      "type": "object",
+      "description": "Job specifications for flink clusters in application deployment mode. The specification is immutable even if job properties are changed by calling the RunJob API, please use the ListJob API to get the latest job information.",
+      "required": [
+        "jobJarDirectory",
+        "jarName",
+        "upgradeMode"
+      ],
+      "properties": {
+        "jobJarDirectory": {
+          "description": "A string property that specifies the directory where the job JAR is located.",
+          "type": "string"
+        },
+        "jarName": {
+          "description": "A string property that represents the name of the job JAR.",
+          "type": "string"
+        },
+        "entryClass": {
+          "description": "A string property that specifies the entry class for the Flink job. If not specified, the entry point is auto-detected from the flink job jar package.",
+          "type": "string"
+        },
+        "args": {
+          "description": "A string property representing additional JVM arguments for the Flink job. It should be space separated value.",
+          "type": "string"
+        },
+        "savePointName": {
+          "description": "A string property that represents the name of the savepoint for the Flink job",
+          "type": "string"
+        },
+        "upgradeMode": {
+          "description": "A string property that indicates the upgrade mode to be performed on the Flink job. It can have one of the following enum values => STATELESS_UPDATE, SAVEPOINT, LAST_STATE_UPDATE.",
```

Comment:
```suggestion
          "description": "A string property that indicates the upgrade mode to be performed on the Flink job. It can have one of the following enum values => STATELESS_UPDATE, UPDATE, LAST_STATE_UPDATE.",
```